From 2977b3c5bdc2a1ef734202df3c253a95c6fe6286 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Wed, 2 Oct 2019 18:17:21 +0200
Subject: [PATCH] boxpointer: Ungrab pointer when hiding

Otherwise we'll continue to receive pointer events, such as click and
motion, while showing the hiding animation.
---
 js/ui/boxpointer.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/js/ui/boxpointer.js b/js/ui/boxpointer.js
index 47f718a84..de24daf52 100644
--- a/js/ui/boxpointer.js
+++ b/js/ui/boxpointer.js
@@ -124,6 +124,10 @@ var BoxPointer = new Lang.Class({
         if (!this.actor.visible)
             return;
 
+        let pointerGrab = Clutter.get_pointer_grab();
+        if (pointerGrab && this.actor.contains(pointerGrab))
+            Clutter.ungrab_pointer();
+
         let xOffset = 0;
         let yOffset = 0;
         let themeNode = this.actor.get_theme_node();
-- 
2.23.0

